docs/deployment-guide-terraform.md

```diff
@@ -0,0 +1,424 @@
+
+<!-- markdownlint-disable MD013 -->
+> NOTE: Azure Cloud Shell is often our preferred place to deploy from because the AZ CLI and Terraform are already installed. However, sometimes Cloud Shell has different versions of the dependencies from what we have tested and verified, and sometimes there have been bugs in the Terraform Azure RM provider or the AZ CLI that only appear in Cloud Shell. If you are deploying from Azure Cloud Shell and see something unexpected, try the [development container](../.devcontainer) or deploy from your machine using locally installed AZ CLI and Terraform. We welcome all feedback and [contributions](../CONTRIBUTING.md), so if you see something that doesn't make sense, please [create an issue](https://github.com/Azure/missionlz/issues/new/choose) or open a [discussion thread](https://github.com/Azure/missionlz/discussions).
+<!-- markdownlint-enable MD013 -->
+
+## Planning
+
+### Decide on a Resource Prefix
+
+Resource Groups and resource names are derived from the `resourcePrefix` parameter, which defaults to 'mlz'. Pick a unqiue resource prefix that is 3-10 alphanumeric characters in length without whitespaces.
+
+### One Subscription or Multiple
+
+MLZ can deploy to a single subscription or multiple subscriptions. A test and evaluation deployment may deploy everything to a single subscription, and a production deployment may place each tier into its own subscription.
+
+The optional parameters related to subscriptions are below. At least one subscription is required.
+
+Parameter name | Default Value | Description
+-------------- | ------------- | -----------
+`hub_subid` | '' | Subscription ID for the Hub deployment
+`tier0_subid` | value of hub_subid | Subscription ID for tier 0
+`tier1_subid` | value of hub_subid | Subscription ID for tier 1
+`tier2_subid` | value of hub_subid | Subscription ID for tier 2
+
+### Networking
+
+The following parameters affect networking addresses. Additional networking parameters can be found in the [`src/terraform/mlz/variables.tf`](../src/terraform/mlz/variables.tf) file.
+
+Parameter name | Default Value | Description
+-------------- | ------------- | -----------
+`hub_vnet_address_space` | '10.0.100.0/24' | The address space to be used for the virtual network
+`hub_client_address_space` | '10.0.100.0/26' | The address space to be used for the Firewall virtual network
+`hub_management_address_space` | '10.0.100.64/26' | The address space to be used for the Firewall virtual network subnet used for management traffic
+`tier0_vnet_address_space` | '10.0.110.0/26' | VNet prefix for tier 0
+`tier0_subnets`.address_prefixes | '10.0.110.0/27' | Subnet prefix for tier 0
+`tier1_vnet_address_space` | '10.0.115.0/26' | VNet prefix for tier 1
+`tier1_subnets`.address_prefixes | '10.0.115.0/27' | Subnet prefix for tier 1
+`tier2_vnet_address_space` | '10.0.120.0/26' | VNet prefix for tier 2
+`tier2_subnets`.address_prefixes | '10.0.120.0/27' | Subnet prefix for tier 2
```

Comment:
I don't know if Terraform supports assigning a value to a single property like `address_prefies` on a map of object like `tier0_subnets` -- does it?

https://github.com/Azure/missionlz/blob/9099c3e83e930620ec6ebe8381b4a7e666dddd39/src/terraform/mlz/variables.tf#L338-L340

Can someone specify `-var="tier0_subnets.address_prefixes=foo"`?

If this is possible, great, we'd just need to fix these cells in the table:

```suggestion
`tier0_subnets.address_prefixes` | '10.0.110.0/27' | Subnet prefix for tier 0
`tier1_vnet_address_space` | '10.0.115.0/26' | VNet prefix for tier 1
`tier1_subnets.address_prefixes` | '10.0.115.0/27' | Subnet prefix for tier 1
`tier2_vnet_address_space` | '10.0.120.0/26' | VNet prefix for tier 2
`tier2_subnets.address_prefixes` | '10.0.120.0/27' | Subnet prefix for tier 2
```

If it's not possible, would we document how to modify the parameter like we do below for `create_policy_assignment`?

Reply:
I added instructions above the table saying to edit the `variables.tf` files to override the defaults.